src/lib/syntax/lexer.rs

```diff
@@ -143,18 +143,18 @@ impl<'a> Lexer<'a> {
     }
 
     /// next fetches the next token and return it, or a LexerError if there are no more.
-    fn next(&mut self) -> Result<char, LexerError> {
+    fn next(&mut self) -> char {
         match self.buffer.next() {
-            Some(ch) => Ok(ch),
-            None => Err(LexerError::new("finished")),
+            Some(ch) => ch,
+            None => panic!(),
```

Comment:
Could we put a message in here?
Like "More more characters to consume from input stream"

Reply:
Done. I also tested (on single line comment and on `/*asdads`) that when this is not satisfied that it panics correctly.